### PR TITLE
Fix trading screen by adding authenticateIfNeeded

### PR DESCRIPTION
**Pull request by jsom**

### CODE REVIEW

File: TradeItIosTicketSDK2/TradeItTradingTicketViewController.swift [hunk 1]
```diff
@@ -25,16 +25,33 @@ class TradeItTradingTicketViewController: TradeItViewController, TradeItSymbolSe
...
+            })
+        }, onSecurityQuestion: { securityQuestion, answerSecurityQuestion, cancelQuestion in
+            self.alertManager.promptUserToAnswerSecurityQuestion(
+                securityQuestion,
+                onViewController: self,
+                onAnswerSecurityQuestion: answerSecurityQuestion,
+                onCancelSecurityQuestion: cancelQuestion
+            )
+        }, onFailure: { errorResult in
+            print(errorResult)
```
Comment (mitochondrion) [inline TradeItIosTicketSDK2/TradeItTradingTicketViewController.swift]: Something needs to happen here:
- take user to account selection screen?
- show alert with error?
- relink if necessary?
- retry?  This is "if needed" so maybe authentication session just expired...

File: TradeItIosTicketSDK2/TradeItTradingTicketViewController.swift [hunk 2]
```diff
@@ -25,16 +25,33 @@ class TradeItTradingTicketViewController: TradeItViewController, TradeItSymbolSe
...
         }
         
         orderActionSelected(orderAction: TradeItOrderActionPresenter.labelFor(order.action))
         orderTypeSelected(orderType: TradeItOrderPriceTypePresenter.labelFor(order.type))
         orderExpirationSelected(orderExpiration: TradeItOrderExpirationPresenter.labelFor(order.expiration))
-        
-        updateSymbolView()
-        updateTradingBrokerAccountView()
+
+        linkedBrokerAccount.linkedBroker.authenticateIfNeeded(onSuccess: {
```
Comment (mitochondrion) [inline TradeItIosTicketSDK2/TradeItTradingTicketViewController.swift]: We should show a spinner somewhere...

File: ExampleApp/ExampleViewController.swift [hunk 3]
```diff
@@ -144,8 +144,8 @@ class ExampleViewController: UIViewController, UITableViewDataSource, UITableVie
         guard let broker = TradeItLauncher.linkedBrokerManager.linkedBrokers.first else { return print("You must link a broker first.") }
         guard let account = broker.accounts.first else { return print("Accounts is empty. Call authenticate on the broker first.") }
 
-        account.getAccountOverview(onSuccess: {
-            print(account.balance ?? "Something went wrong!")
+        account.getAccountOverview(onSuccess: { balance in
+            print(balance ?? "Something went wrong!")
```
Comment (mitochondrion) [inline ExampleApp/ExampleViewController.swift]: Is this `print` for debugging?  Remove if no longer needed.

File: TradeItIosTicketSDK2/TradeItLinkedBroker.swift [hunk 4]
```diff
@@ -94,7 +94,9 @@ import PromiseKit
     public func refreshAccountBalances(onFinished: @escaping () -> Void) {
         let promises = accounts.map { account in
             return Promise<Void> { fulfill, reject in
-                account.getAccountOverview(onSuccess: fulfill, onFailure: { errorResult in
+                account.getAccountOverview(onSuccess: { _ in
```
Comment (mitochondrion) [inline TradeItIosTicketSDK2/TradeItLinkedBroker.swift]: This just seems more verbose.  What's the purpose of this change?

File: TradeItIosTicketSDK2/TradeItTradingTicketViewController.swift [hunk 5]
```diff
@@ -307,15 +325,25 @@ class TradeItTradingTicketViewController: TradeItViewController, TradeItSymbolSe
             , let linkedBrokerAccount = order.linkedBrokerAccount
             else { return }
 
-        linkedBrokerAccount.getPositions(onSuccess: { positions in
-            let positionsMatchingSymbol = positions.filter { portfolioPosition in
-                TradeItPortfolioPositionPresenterFactory.forTradeItPortfolioPosition(portfolioPosition).getFormattedSymbol() == symbol
-            }
+        linkedBrokerAccount.linkedBroker.authenticateIfNeeded(onSuccess: {
```
Comment (mitochondrion) [inline TradeItIosTicketSDK2/TradeItTradingTicketViewController.swift]: Aren't we now already authenticating when the screen loads?

File: TradeItIosTicketSDK2/TradeItTradingTicketViewController.swift (hunk 1 above)
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItTradingTicketViewController.swift]: Discuss in person.

File: ExampleApp/ExampleViewController.swift (hunk 3 above)
Comment (jsom) [inline ExampleApp/ExampleViewController.swift]: It's the example app - it's to show partners how they can retrieve the information.

File: TradeItIosTicketSDK2/TradeItLinkedBroker.swift (hunk 4 above)
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItLinkedBroker.swift]: You asked for `getAccountOverview` to pass in the account overview... you have to handle the new parameter. `fulfill` takes no parameters, hence you can't just set `onSuccess: fulfill`.
